protocol/rpcprovider/rpcprovider_server.go

```diff
@@ -323,21 +327,49 @@ func (rpcps *RPCProviderServer) verifyRelaySession(ctx context.Context, request
 		return nil, nil, utils.LavaFormatWarning("did not pass relay validation", err, utils.Attribute{Key: "GUID", Value: ctx})
 	}
 	// check signature
-	extractedConsumerAddress, err = sigs.ExtractSignerAddress(request.RelaySession)
-	if err != nil {
-		return nil, nil, utils.LavaFormatWarning("extract signer address from relay", err, utils.Attribute{Key: "GUID", Value: ctx})
-	}
+	var consumerAddressString string
 
-	// handle non data reliability relays
-	consumerAddressString := extractedConsumerAddress.String()
+	if request.RelaySession.Badge != nil {
```

Comment:
extract this entire if case body into a function and call it validateBadgeSession

Reply:
Added the function & extracted the validation process in this commit: 35f104b23ba6af6528966a40fad93659e310ea49